vita3k/modules/SceNpTrophy/SceNpTrophy.cpp

```diff
@@ -197,9 +197,89 @@ EXPORT(int, sceNpTrophyGetGroupIcon) {
+        || (api_handle == -1)
+        || ((group_id < -1) || (group_id >= np::trophy::MAX_GROUPS))
+        || (!details && !data)
+        || (details && (details->size != sizeof(*details)))
+        || (data && (data->size != sizeof(*data)))) {
+        return RET_ERROR(SCE_NP_TROPHY_ERROR_INVALID_ARGUMENT);
+    }
+
+    np::trophy::Context *context = get_trophy_context(emuenv.np.trophy_state, context_handle);
+    if (!context) {
+        return RET_ERROR(SCE_NP_TROPHY_ERROR_INVALID_CONTEXT);
+    }
+
+    if (group_id != -1) { // SCE_NP_TROPHY_BASE_GAME_GROUP_ID
+        STUBBED("trophies groups are not supported");
+        return UNIMPLEMENTED();
+    }
+
+    if (details) {
+        details->groupId = group_id;
+        details->numTrophies = context->trophy_count;
+
+        std::string name, detail;
+        if (!context->get_trophy_set(name, detail))
+            return RET_ERROR(SCE_NP_TROPHY_ERROR_UNSUPPORTED_TROPHY_CONF);
+
+        memcpy((char *)details->title, name.c_str(), name.size() + 1);
+        memcpy((char *)details->description, detail.c_str(), detail.size() + 1);
+
+        for (std::uint32_t i = 0; i < context->trophy_count; i++) {
+            switch (context->trophy_kinds[i]) {
+            case np::trophy::SceNpTrophyGrade::SCE_NP_TROPHY_GRADE_PLATINUM:
+                details->numPlatinum++;
+                break;
+            case np::trophy::SceNpTrophyGrade::SCE_NP_TROPHY_GRADE_GOLD:
+                details->numGold++;
+                break;
+            case np::trophy::SceNpTrophyGrade::SCE_NP_TROPHY_GRADE_SILVER:
+                details->numSilver++;
+                break;
+            case np::trophy::SceNpTrophyGrade::SCE_NP_TROPHY_GRADE_BRONZE:
+                details->numBronze++;
+                break;
+            default:
+                break;
+            }
+        }
+    }
+
+    if (data) {
+        data->unlockedTrophies = context->total_trophy_unlocked();
+
+        for (std::uint32_t i = 0; i < context->trophy_count; i++) {
+            if (context->is_trophy_unlocked(i)) {
+                switch (context->trophy_kinds[i]) {
+                case np::trophy::SceNpTrophyGrade::SCE_NP_TROPHY_GRADE_PLATINUM:
+                    data->unlockedPlatinum++;
+                    break;
+                case np::trophy::SceNpTrophyGrade::SCE_NP_TROPHY_GRADE_GOLD:
+                    data->unlockedGold++;
+                    break;
+                case np::trophy::SceNpTrophyGrade::SCE_NP_TROPHY_GRADE_SILVER:
+                    data->unlockedSilver++;
+                    break;
+                case np::trophy::SceNpTrophyGrade::SCE_NP_TROPHY_GRADE_BRONZE:
+                    data->unlockedBronze++;
+                    break;
+                default:
+                    break;
+                }
+            }
+        }
+
+        data->progressPercentage = data->unlockedTrophies * 100 / context->trophy_count;
+    }
+    return STUBBED("");
```

Comment:
Why the STUBBED("") ?

Reply:
Cause it's still stubbed, but I rewrote STUBBED message to explain this better.